yt_dlp/extractor/linkedin.py

```diff
@@ -83,42 +82,69 @@ def _get_video_id(self, video_data, course_slug, video_slug):
-            'title': 'Mishal K. on LinkedIn: #sendinblueviews #toronto #digitalmarketing',
-            'description': 'md5:be125430bab1c574f16aeb186a4d5b19',
-            'creator': 'Mishal K.'
+            'title': 'Mishal K. on LinkedIn: #sendinblueviews #toronto #digitalmarketing #nowhiring #sendinblue…',
+            'description': 'md5:2998a31f6f479376dd62831f53a80f71',
+            'uploader': 'Mishal K.',
+            'thumbnail': 'https://media.licdn.com/dms/image/C4E05AQG7hCp7zIeciw/feedshare-thumbnail_720_1280/0/1633381554858?e=2147483647&v=beta&t=teGgAD-Z8A2K4UrwAtEgHz9xGMrRLmUK6cE3vm3csK0',
+            'like_count': int
+        },
+    }, {
+        'url': 'https://www.linkedin.com/posts/the-mathworks_2_what-is-mathworks-cloud-center-activity-7151241570371948544-4Gu7',
+        'info_dict': {
+            'id': 'D5610AQFKo9M0zqY2_g',
+            'display_id': 'the-mathworks_2_what-is-mathworks-cloud-center-activity-7151241570371948544-4Gu7',
+            'ext': 'mp4',
+            'title': 'MathWorks on LinkedIn: What Is MathWorks Cloud Center?',
+            'description': 'md5:95f9d4eeb6337882fb47eefe13d7a40c',
+            'uploader': 'MathWorks',
+            'thumbnail': 'https://media.licdn.com/dms/image/D5610AQFKo9M0zqY2_g/ads-video-thumbnail_720_1280/0/1704988806715?e=2147483647&v=beta&t=0vg-ksOY2KrL_QFlNacCv5Tmuk0tum9FJ_4dlJ56Gyw',
+            'like_count': int,
+            'subtitles': 'mincount:1'
         },
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
 
-        title = self._html_extract_title(webpage)
-        description = clean_html(get_element_by_class('share-update-card__update-text', webpage))
-        like_count = int_or_none(get_element_by_class('social-counts-reactions__social-counts-numRections', webpage))
-        creator = strip_or_none(clean_html(get_element_by_class('comment__actor-name', webpage)))
-
-        sources = self._parse_json(extract_attributes(self._search_regex(r'(<video[^>]+>)', webpage, 'video'))['data-sources'], video_id)
+        title = self._og_search_title(webpage, default=None) or self._html_extract_title(webpage)
+        description = self._og_search_description(webpage, default=None)
+        json_ld_list = list(self._yield_json_ld(webpage, video_id))
+        uploader = traverse_obj(
+            json_ld_list, (lambda _, v: v['@type'] == 'SocialMediaPosting', 'author'), get_all=False).get('name')
+        like_count = int_or_none(self._search_regex(r'<a href=[^>]+\sdata-num-reactions="(\d+)"', webpage, 'reactions', default=None))
+
+        video_attrs = extract_attributes(self._search_regex(r'(<video[^>]+>)', webpage, 'video'))
+        print('video_attrs:', video_attrs)
+        sources = self._parse_json(video_attrs['data-sources'], video_id)
+        id = str(video_attrs.get('data-digitalmedia-asset-urn') or video_id).replace('urn:li:digitalmediaAsset:', '')
```

Comment:
Is there a need to change the id from the numeric id in the URL? are there posts URLs without a numeric id? I can't seem to find any. IMO we should keep the id the same as it was before.

If there is a good reason to change it, though, then:
- don't redefine built-ins / don't use `id` as var name
- we still need to try and match the numeric id from the URL and return it in the info dict under the `_old_archive_ids` field

Reply:
I cannot explain a good reason, except that the video_attrs JSON has that asset URN.

I don't understand the documented explanation of `_old_archive_ids`. What it its use, and what is it supposed to contain? Before my change, extracting the `id` was buggy and returned an arbitrary intermediate digit (`2`). Or is it meant to follow the original extraction scheme, seeing the site's URL scheme (probably) hasn't changed?

Either way, I will stick to the old `id` with its fixed extraction.

Reply:
If we change what we extract as the `id` for an extractor, we need to try and preserve the old id values so that user's download archives for the site are not invalidated. `_old_archive_ids` is a list of old ids for the video that will be matched against a user's download archive to see if it has already been downloaded. You would use the `make_archive_id` utils function to format it properly, e.g.:

```py
        '_old_archive_ids': [make_archive_id(LinkedInIE, old_video_id)],
```

but since we are keeping the old id format, this suggestion is no longer needed